source/agora/common/DNS.d

```diff
@@ -359,7 +359,10 @@ public struct Header
         return cast(RCode) val;
     }
     /// Ditto
-    public void RCODE (RCode val) scope { this.field2 |= val; }
+    public void RCODE (RCode val) scope
+    {
+        this.field2 = (0xfff0 & this.field2) | val;
+    }
```

Comment:
How is that different from the previous code though ? `val.max == 5`, which fits in 4 bits.

Reply:
It was not possible to return back to `NoError` (0) once a return code is set. Or setting `FormatError` first and `ServerFailure` later would result in `NameError`

Reply:
Missed that. Makes perfect sense.